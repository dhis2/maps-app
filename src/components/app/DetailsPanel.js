import cx from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'
import { useSelector } from 'react-redux'
import Interpretations from '../interpretations/Interpretations.js'
import OrgUnitProfile from '../orgunits/OrgUnitProfile.js'
import styles from './styles/DetailsPanel.module.css'

const DetailsPanel = ({ interpretationsRenderCount }) => {
    const detailsPanelOpen = useSelector((state) => state.ui.rightPanelOpen)
    const viewOrgUnitProfile = useSelector((state) => state.orgUnitProfile)
    const interpretationId = useSelector((state) => state.interpretation?.id)

    const getContent = () => {
        if (interpretationId || (detailsPanelOpen && !viewOrgUnitProfile)) {
            return <Interpretations renderCount={interpretationsRenderCount} />
        }

        return detailsPanelOpen ? <OrgUnitProfile /> : null
    }

    return (
        <div
            className={cx(styles.detailsPanel, {
                [styles.collapsed]: !detailsPanelOpen,
            })}
        >
            {getContent()}
        </div>
    )
}

DetailsPanel.propTypes = {
    interpretationsRenderCount: PropTypes.number.isRequired,
}

export default DetailsPanel
