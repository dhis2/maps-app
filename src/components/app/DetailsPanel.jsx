import cx from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'
import { useSelector } from 'react-redux'
import FeatureProfile from '../feature/FeatureProfile.jsx'
import Interpretations from '../interpretations/Interpretations.jsx'
import OrgUnitProfile from '../orgunits/OrgUnitProfile.jsx'
import styles from './styles/DetailsPanel.module.css'

const DetailsPanel = ({ interpretationsRenderCount }) => {
    const detailsPanelOpen = useSelector((state) => state.ui.rightPanelOpen)
    const viewOrgUnitProfile = useSelector((state) => state.orgUnitProfile)
    const viewFeatureProfile = useSelector((state) => !!state.featureProfile)
    const interpretationId = useSelector((state) => state.interpretation?.id)

    const getContent = () => {
        if (
            interpretationId ||
            (detailsPanelOpen && !viewOrgUnitProfile && !viewFeatureProfile)
        ) {
            return <Interpretations renderCount={interpretationsRenderCount} />
        }

        if (detailsPanelOpen) {
            if (viewOrgUnitProfile) {
                return <OrgUnitProfile />
            }
            if (viewFeatureProfile) {
                return <FeatureProfile />
            }
        }

        return null
    }

    return (
        <div
            data-test="details-panel"
            className={cx(styles.detailsPanel, {
                [styles.collapsed]: !detailsPanelOpen,
            })}
        >
            {getContent()}
        </div>
    )
}

DetailsPanel.propTypes = {
    interpretationsRenderCount: PropTypes.number.isRequired,
}

export default DetailsPanel
