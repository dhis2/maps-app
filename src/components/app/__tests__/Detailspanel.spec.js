import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import DetailsPanel from '../DetailsPanel.js'

const mockStore = configureMockStore()

jest.mock(
    '../../interpretations/Interpretations.js',
    () =>
        function MockInterpretations() {
            return <div>Interpretations</div>
        }
)

jest.mock(
    '../../orgunits/OrgUnitProfile.js',
    () =>
        function MockOrgUnitProfile() {
            return <div>Org Unit Profile</div>
        }
)

describe('DetailsPanel', () => {
    test('renders InterpretationsPanel when has interpretationId, has orgUnitProfile and panel is open', () => {
        const store = {
            interpretation: { id: 'abc123' },
            orgUnitProfile: 'xyzpdq',
            ui: { rightPanelOpen: true },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <DetailsPanel interpretationsRenderCount={1} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('renders InterpretationsPanel when has interpretationId, has orgUnitProfile and panel is closed', () => {
        const store = {
            interpretation: { id: 'abc123' },
            orgUnitProfile: 'xyzpdq',
            ui: { rightPanelOpen: false },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <DetailsPanel interpretationsRenderCount={1} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('renders InterpretationsPanel when has interpretationId, no orgUnitProfile and panel is open', () => {
        const store = {
            interpretation: { id: 'abc123' },
            orgUnitProfile: null,
            ui: { rightPanelOpen: true },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <DetailsPanel interpretationsRenderCount={1} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })

    test('renders InterpretationsPanel when has interpretationId, no orgUnitProfile and panel is closed', () => {
        const store = {
            interpretation: { id: 'abc123' },
            orgUnitProfile: null,
            ui: { rightPanelOpen: false },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <DetailsPanel interpretationsRenderCount={1} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })

    test('renders OrgUnitProfile when no interpretationId, has orgUnitProfile and panel is open', () => {
        const store = {
            interpretation: {},
            orgUnitProfile: 'xyzpdq',
            ui: { rightPanelOpen: true },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <DetailsPanel interpretationsRenderCount={1} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })

    test('renders null when no interpretationId, has orgUnitProfile, and panel closed', () => {
        const store = {
            interpretation: {},
            orgUnitProfile: 'xyzpdq',
            ui: { rightPanelOpen: false },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <DetailsPanel interpretationsRenderCount={1} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('renders InterpretationsPanel when no interpretationId, no orgUnitProfile and panel open', () => {
        const store = {
            interpretation: {},
            orgUnitProfile: null,
            ui: { rightPanelOpen: true },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <DetailsPanel interpretationsRenderCount={1} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })

    test('renders null when no interpretationId, no orgUnitProfile, and panel closed', () => {
        const store = {
            interpretation: {},
            orgUnitProfile: null,
            ui: { rightPanelOpen: false },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <DetailsPanel interpretationsRenderCount={1} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
})
