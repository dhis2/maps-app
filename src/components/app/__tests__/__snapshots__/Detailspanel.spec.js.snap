// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPanel renders FeatureProfile when interpretation=undefined, orgUnitProfile=null, featureProfile=myfeature and panelOpen=true 1`] = `
<div>
  <div
    class="detailsPanel"
    data-test="details-panel"
  >
    <div>
      Feature Profile
    </div>
  </div>
</div>
`;

exports[`DetailsPanel renders Interpretations when interpretation=myinterpretation, orgUnitProfile=myorgunit, featureProfile=myfeature and panelOpen=false 1`] = `
<div>
  <div
    class="detailsPanel collapsed"
    data-test="details-panel"
  >
    <div>
      Interpretations
    </div>
  </div>
</div>
`;

exports[`DetailsPanel renders Interpretations when interpretation=myinterpretation, orgUnitProfile=myorgunit, featureProfile=myfeature and panelOpen=true 1`] = `
<div>
  <div
    class="detailsPanel"
    data-test="details-panel"
  >
    <div>
      Interpretations
    </div>
  </div>
</div>
`;

exports[`DetailsPanel renders Interpretations when interpretation=myinterpretation, orgUnitProfile=null, featureProfile=null and panelOpen=false 1`] = `
<div>
  <div
    class="detailsPanel collapsed"
    data-test="details-panel"
  >
    <div>
      Interpretations
    </div>
  </div>
</div>
`;

exports[`DetailsPanel renders Interpretations when interpretation=myinterpretation, orgUnitProfile=null, featureProfile=null and panelOpen=true 1`] = `
<div>
  <div
    class="detailsPanel"
    data-test="details-panel"
  >
    <div>
      Interpretations
    </div>
  </div>
</div>
`;

exports[`DetailsPanel renders Interpretations when interpretation=undefined, orgUnitProfile=null, featureProfile=null and panelOpen=true 1`] = `
<div>
  <div
    class="detailsPanel"
    data-test="details-panel"
  >
    <div>
      Interpretations
    </div>
  </div>
</div>
`;

exports[`DetailsPanel renders OrgUnitProfile when interpretation=undefined, orgUnitProfile=myorgunit, featureProfile=myfeature and panelOpen=true 1`] = `
<div>
  <div
    class="detailsPanel"
    data-test="details-panel"
  >
    <div>
      Org Unit Profile
    </div>
  </div>
</div>
`;

exports[`DetailsPanel renders null when interpretation=undefined, orgUnitProfile=myorgunit, featureProfile=null and panelOpen=false 1`] = `
<div>
  <div
    class="detailsPanel collapsed"
    data-test="details-panel"
  />
</div>
`;

exports[`DetailsPanel renders null when interpretation=undefined, orgUnitProfile=null, featureProfile=null and panelOpen=false 1`] = `
<div>
  <div
    class="detailsPanel collapsed"
    data-test="details-panel"
  />
</div>
`;
